hera_cal/lstbin.py

```diff
@@ -579,110 +577,137 @@ def lst_bin_files(data_files, filetype='uvh5', input_cals=None, dlst=None, verbo
     kwargs['outdir'] = outdir
     kwargs['overwrite'] = overwrite
 
-    # get metadata from the zeroth data file
-    d, fl, ap, a, f, t, l, p = io.load_vis(data_files[0][0], return_meta=True, pick_data_ants=False, filetype=filetype)
+    # get metadata from the zeroth data file in the last day
+    uv = UVData()
+    uv.read_uvh5(data_files[-1][0], read_data=False)
 
     # get frequency, time (converting times in days to integration times in seconds) and antenna position information
-    freq_array = copy.copy(f)
-    antpos = copy.deepcopy(ap)
-    start_jd = np.floor(t)[0]
+    freq_array = uv.freq_array[0].copy()
+    antpos, ants = uv.get_ENU_antpos(center=True)
+    antpos = dict(zip(ants, antpos))
+    times = np.unique(uv.time_array)
+    start_jd = np.floor(times.min())
     kwargs['start_jd'] = start_jd
-    integration_time = np.median(np.diff(t)) * 24 * 3600.
-    assert np.all(np.abs(np.diff(t) - np.median(np.diff(t))) < 1e-6), 'All integrations must be of equal length (BDA not supported).'
-    del d, fl, ap, a, f, t, l, p
-    garbage_collector.collect()
+    integration_time = np.median(np.diff(times)) * 24 * 3600.
+    assert np.all(np.abs(np.diff(times) - np.median(np.diff(times))) < 1e-6), 'All integrations must be of equal length (BDA not supported).'
+
+    # get baselines from data and form baseline groups
+    bls = sorted(uv.get_antpairs())
+    Nbls = len(bls)
+    blgroups = [bls[i * (Nbls // Nblgroups + 1):(i + 1) * (Nbls // Nblgroups + 1)] for i in range(Nblgroups)]
```

Comment:
Does this break if Nblgroups is greater than the number of baselines? If you end up using `nbl_to_load`, you can see how I handled this problem here: https://github.com/HERA-Team/hera_cal/blob/ed46b2c546831259c62bc187240562db2ec9f45c/hera_cal/delay_filter.py#L166